### PR TITLE
Bootstrapping Apollo Server and Queries

### PR DESCRIPTION
**Pull request by prnvpndy**
**Description**

AC

Create an Apollo Server that will have a getMyProfile query in its schema and querying which will return your name, id, and email address as a response.

**Prerequisites**
**Install below packages as dev dependencies**
- @babel/core
- @babel/node
- @babel/preset-env
- nodemon

**Install below packages as direct dependencies**
- apollo-server-express
- express
- graphql
- merge-graphql-schemas

Add and configure babel and eslint just like you did in express application.

Create src folder and add the below files and folder.

- src/modules/user/query.js
- src/modules/user/types.graphql
- src/modules/user/index.js
- src/modules/schema.graphql
- src/modules/index.js
- src/index.js
- src/server.js
- src/config/configurations.js

**Steps**

    Add type for User in user/types.graphql file and define schema for id, name, and email address.
    Add type query getMyProfile in modules/schema.graphql file and add its return type as User.
    Add resolver getMyProfile in user/query.js file and return your information (name, email and id).
    Combine all of the graphql file to create a schema.
        Create an index file in src/modules folder and import fileLoader and mergeTypes from merge-graphql-schemas.
        With fileLoader get all the .graphl files and merge them with the help of mergeTypes.
        import the user module's query and export a single object with typeDefs and resolver. 
    Configure the server just like you did in express application.
        Create an extra function called setupApollo, where you will create an instance of apollo server and you will add that as middleware to your express application
    Add the config folder and export your environment variables.
    Bootstrap the express server, call setupApollo and start the server.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PORT=9001
+NODE_ENV=dev
```
Comment (mohd-aazam) [inline .env]: don't push .env, keep this in .gitignore

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import Server from './Server';
+import config from './config/configuration';
+import schema from './modules';
+
+const server = new Server(config);
+(() => {
+  server.bootstrap().setupApollo(schema);
+})();
```
Comment (mohd-aazam) [inline src/index.js]: Any particular reason to create an immediate function call

**Review by mohd-aazam — CHANGES_REQUESTED** [review]

**Comment by mohd-aazam** [comment]
Remove package-lock.json file, add into .gitignore

File: src/index.js (hunk 2 above)
Comment (prnvpndy) [reply src/index.js]: I thought this function will not run without, arrow function, but its working fine when i removed arrow function. I have updated the code.

**Pull request merged**